app/src/main/java/com/cliqz/browser/utils/HttpHandler.java

```diff
@@ -1,70 +1,123 @@
 package com.cliqz.browser.utils;
 
+import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 import android.util.Log;
 
+import com.cliqz.utils.StreamUtils;
+
 import org.json.JSONException;
 import org.json.JSONObject;
 
-import java.io.BufferedReader;
-import java.io.DataInputStream;
-import java.io.DataOutputStream;
 import java.io.IOException;
-import java.io.InputStreamReader;
+import java.io.InputStream;
+import java.io.OutputStream;
 import java.net.HttpURLConnection;
+import java.net.ProtocolException;
 import java.net.URL;
 
 /**
  * @author Khaled Tantawy
+ * @author Stefano Pacifici
  */
-
 public class HttpHandler {
 
     private static final String HEADER_CONTENT_TYPE = "Content-Type";
 
     private static final String TAG = HttpHandler.class.getSimpleName();
-
+    private static final String HTTP_METHOD_GET = "GET";
+    private static final int HTTP_CONNECT_TIMEOUT = 10000; // ms
+    private static final int HTTP_READ_TIMEOUT = HTTP_CONNECT_TIMEOUT;
+    private static final int HTTP_CODE_OK = 200;
```

Comment:
There's a constant in the HttpURLConnection class.

 `HttpURLConnection.HTTP_OK`